### PR TITLE
Add data set id fields

### PR DESCRIPTION
**Pull request by exyi**
Adds dataSetId Read/Insert/Upsert schemas of the data source.

I have only fixed existing tests to count with the `dataSetId` field, not added any new tests. I think the existing tests cover well that the mapping actually works.

Note that filtering based on `dataSetId` is not supported, I don't know if that's something we'd like to have.

### CODE REVIEW

**Comment by exyi** [comment]
> I would like to see some tests that attempt to read the resources back again, and assert that the dataSetId matches what you provide during creation.

Sure, I have added a dataSet into the `jetfiretest2` db and reused Asset test to check that the dataSetId is inserted correctly.

File: src/test/scala/cognite/spark/v1/AssetsRelationTest.scala [hunk 1]
```diff
@@ -145,9 +146,13 @@ class AssetsRelationTest extends FlatSpec with Matchers with SparkTest {
     val assetsCreated = getNumberOfRowsCreated(metricsPrefix, "assets")
     assert(assetsCreated == 1)
 
-    retryWhile[Array[Row]](
+    val Array(createdAsset) = retryWhile[Array[Row]](
```
Comment (sebbert) [inline src/test/scala/cognite/spark/v1/AssetsRelationTest.scala]: Pro tip: You can use [`loneElement`](http://doc.scalatest.org/3.0.1-2.12/org/scalatest/LoneElement.html) to get better error messages when you expect a collection to contain a single element. I don't consider that a blocker for this PR though.

File: src/test/scala/cognite/spark/v1/AssetsRelationTest.scala [hunk 2]
```diff
@@ -145,9 +146,13 @@ class AssetsRelationTest extends FlatSpec with Matchers with SparkTest {
...
       spark.sql(s"select * from assets where source = '$assetsTestSource'").collect,
       rows => rows.length < 1)
+
+    assert(createdAsset.getAs[String]("name") == "asset name")
+    assert(createdAsset.getAs[Long]("dataSetId") == testDataSetId)
+    assert(createdAsset.getAs[String]("externalId") == "1")
```
Comment (sebbert) [inline src/test/scala/cognite/spark/v1/AssetsRelationTest.scala]: We should use the `x should be(y)` or `x shouldBe y` matchers when asserting equality.

```suggestion
    createdAsset.getAs[String]("name") shouldBe "asset name"
    createdAsset.getAs[Long]("dataSetId") shouldBe testDataSetId
    createdAsset.getAs[String]("externalId") shouldBe "1"
```

Comment (sebbert) [reply src/test/scala/cognite/spark/v1/AssetsRelationTest.scala]: We should also perform the same checks for other resource types (events, time series, etc.).

Comment (exyi) [reply src/test/scala/cognite/spark/v1/AssetsRelationTest.scala]: If you mean... However, I managed to find a disadvantage of `shouldBe` - when I get the types wrong (for example `.getAs[Option[Long]]("x") == 1`), Scala compiler complain. It just compiles and fails the test when I use use `shouldBe` :/

Comment (sebbert) [reply src/test/scala/cognite/spark/v1/AssetsRelationTest.scala]: Agh, that's a good point. Feel free to disregard this comment then.

Comment (sebbert) [reply src/test/scala/cognite/spark/v1/AssetsRelationTest.scala]: I would still like to see tests for other resource types though :)

**Pull request merged**